### ISSUE TITLE
Conflicting definitions in SPEC.md

### ISSUE THREAD

**Issue by pierwill** [influxdata/flux#5186]
While expressing the Flux specification in PLT Redex (see https://github.com/pierwill/flux-redex), I found two duplicate, conflicting productions.

`Property` is defined as part of a [`RecordLiteral`](https://github.com/influxdata/flux/blob/master/docs/SPEC.md#record-literals). `Parameter` is defined as part of a [`FunctionLiteral`](https://github.com/influxdata/flux/blob/master/docs/SPEC.md#function-literals).

Both `Property` and `Parameter` are also defined (differently) as part of a [`BuiltinStatement`](https://github.com/influxdata/flux/blob/master/docs/SPEC.md#system-built-ins).

This duplication creates ambiguity for readers and for verification. We might consider "namespacing" the items that appear as part of a `BuiltinStatement`, either all of them (e.g. `MonoType-in-builtin`) or just the duplicates (`Parameter-in-builtin`, `Property-in-builtin`).

**Comment by nathanielc** [comment]
Yeah we likely just need to rename the rules for the built in statement. Maybe something like this:


```
BuiltinStatement = "builtin" identifier ":" TypeExpression .
TypeExpression   = MonoType ["where" Constraints] .

MonoType = Tvar | BasicType | ArrayType | RecordType | FunctionType .
Tvar     = "A" … "Z" .
BasicType    = "int" | "uint" | "float" | "string" | "bool" | "time" | "duration" | "bytes" | "regexp" .
ArrayType    = "[" MonoType "]" .
RecordType   = ( "{" [RecordTypeProperties] "}" ) | ( "{" Tvar "with" Properties "}" ) .
FunctionType = "(" [FunctionTypeParameters] ")" "=>" MonoType .

RecordTypeProperties = Property { "," RecordTypeProperty } .
RecordTypeProperty   = Label ":" MonoType .
Label      = identifier | string_lit

FunctionTypeParameters = FunctionTypeParameter { "," FunctionTypeParameter } .
FunctionTypeParameter  = [ "<-" | "?" ] identifier ":" MonoType .

Constraints = Constraint { "," Constraint } .
Constraint  = Tvar ":" Kinds .
Kinds       = identifier { "+" identifier } .
```

Its a bit wordy but makes it clear. Open to other suggestions, but the gist is we need to rename things to avoid the conflict
